runtime-modules/staking-handler/src/lib.rs

```diff
@@ -11,66 +11,86 @@ use frame_support::traits::{Currency, Get, LockIdentifier, LockableCurrency, Wit
 use sp_arithmetic::traits::Zero;
 use sp_std::marker::PhantomData;
 
+// re-export BalanceLock used in LockComparator so that modules using StakingHandler don't have to depend on pallet_balances
+pub use pallet_balances::BalanceLock;
+
 #[cfg(test)]
 mod mock;
 #[cfg(test)]
 mod test;
 
-/// Type alias for member id.
-pub type MemberId<T> = <T as membership::Trait>::MemberId;
-
-/// Balance alias for `balances` module.
-pub type BalanceOf<T> = <T as pallet_balances::Trait>::Balance;
+/// Trait for (dis)allowing certain stake locks combinations.
+pub trait LockComparator<Balance> {
+    /// Checks if stake lock that is about to be used is conflicting with existing locks.
+    fn are_locks_conflicting(
+        new_lock: &LockIdentifier,
+        existing_locks: &[BalanceLock<Balance>],
```

Comment:
Why do you pass the whole BalanceLock instead of just the LockId ?

Reply:
I wanted to give Runtime a possibility to take into account the amount locked (and possibly even the `Reasons`), but maybe it is overengineering.